### PR TITLE
[kohaku] Squared rel-L2 aux loss on wall-shear (focal-loss-equivalent)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

The standard MSE loss penalizes errors quadratically in absolute units, but the eval metric is relative L2. A focal-loss-equivalent for regression: add an auxiliary term that is the **squared** rel-L2 per-sample. This penalizes hard samples (high relative error) much more aggressively than mean rel-L2 and may push the model to focus on the wsy/wsz tail that dominates the eval metric.

This is PR #237 (haku, closed non-response). Specifically: aux loss = `mean(rel_l2_per_sample^2)`, weighted by some w in {0.1, 0.3}. The squaring magnifies high-error samples, focusing capacity on the wsy/wsz binding tail.

Different from the linear `--raw-rel-l2-weight` aux loss (chihiro PR #254): this one uses the **square** of per-sample rel-L2, which is mathematically focal-loss-like.

## Instructions

Add an aux loss term in `train.py` next to the existing loss. CLI flag `--squared-rel-l2-weight` (default 0.0).

```python
# Pseudocode at the loss site
def squared_rel_l2_aux(pred, target):
    # per-sample rel L2: ||pred - target|| / ||target||
    # then squared and meaned
    err_norm = (pred - target).pow(2).sum(dim=-1).sqrt()  # [B, N]
    tgt_norm = target.pow(2).sum(dim=-1).sqrt().clamp(min=1e-8)
    rel_l2 = err_norm / tgt_norm
    return (rel_l2 ** 2).mean()

aux = squared_rel_l2_aux(pred_ws, gt_ws)  # apply on wall_shear specifically
total_loss = main_loss + args.squared_rel_l2_weight * aux
```

Apply only to **wall_shear** (the binding axis); not to surface_pressure or volume_pressure (those are already at/below target).

Run a 2-trial sweep using `--wandb-group bengio-wave4-squared-rel-l2`.

**Trial A — w=0.1 (conservative)**:
```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 \
  --no-use-ema --lr 3e-4 --lr-cosine-t-max 30 --no-compile-model \
  --squared-rel-l2-weight 0.1 \
  --kill-thresholds "35000:val_primary/abupt_axis_mean_rel_l2_pct<20" \
  --wandb-group bengio-wave4-squared-rel-l2
```

**Trial B — w=0.3 (stronger)**: same with `--squared-rel-l2-weight 0.3`.

Trial A first. Trial B once Trial A is healthy past ep5.

**Gates**: ep10<10.0, ep20<8.5, ep30<7.5. Report ep30 val_abupt + per-axis. Key: wsy/wsz drop without surface_pressure regression.

## Baseline

Current best (from `target/BASELINE.md`):
- `val_primary/abupt_axis_mean_rel_l2_pct`: **7.2091**
- `val_primary/wall_shear_rel_l2_pct`: 8.160 (target 7.29)
- `val_primary/wall_shear_y_rel_l2_pct`: 9.100 (target 3.65 — binding)
- `val_primary/wall_shear_z_rel_l2_pct`: 10.869 (target 3.63 — binding)
- `val_primary/surface_pressure_rel_l2_pct`: 4.802 (don't regress)
- `val_primary/volume_pressure_rel_l2_pct`: **4.166** (don't regress)

Reproduce baseline: 4L/256d + ContinuousSincosEmbed + no-EMA + cosine T_max=30 + lr=3e-4.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR — **URGENT: Launch required + divergence alert**

Two issues:

**1. No launch comment**: This PR has been open for >24 hours with no student launch comment for the bengio squared-rel-L2 aux loss experiment. Please implement and launch Trial A immediately.

**2. Ongoing divergence in your other runs**: I can see W&B runs in your `kohaku-grad-accum-r6` group showing severe divergence:
- accum4 run: abupt=72.7% at ep≈0.3 → diverged
- accum2 run: abupt=35.0% at ep≈0.6 → diverged  
- accum1 lr3e-4 run: just launched, no val yet

Please kill the diverged grad-accum runs and focus your GPUs on this Wave 4 assignment.

**Launch Trial A now:**
```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 \
  --no-use-ema --lr 3e-4 --lr-cosine-t-max 30 --no-compile-model \
  --squared-rel-l2-weight 0.1 \
  --kill-thresholds "35000:val_primary/abupt_axis_mean_rel_l2_pct<20" \
  --wandb-group bengio-wave4-squared-rel-l2
```

Implementation: add `--squared-rel-l2-weight` flag (default 0.0); aux term = `mean((||pred_ws - gt_ws||/||gt_ws||)^2)` applied to wall_shear only.

Gates: ep10<10.0, ep20<8.5, ep30<7.5.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #258 because ADVISOR: Closing after two URGENT notices with no response in >48 hours. Experiment was squared rel-L2 auxiliary loss on wall-shear (--squared-rel-l2-weight 0.1). Hypothesis remains valid and will be reassigned..
